addons/addon-base-raas/packages/base-raas-workflow-steps/lib/steps/bulk-reachability-check/bulk-reachability-check.js

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const StepBase = require('@aws-ee/base-workflow-core/lib/workflow/helpers/step-base');
+const { processInBatches } = require('@aws-ee/base-services/lib/helpers/utils');
+
+class BulkReachabilityCheck extends StepBase {
+  async start() {
+    this.print('start pinging data source accounts');
+
+    // Get services
+    const dataSourceReachabilityService = await this.mustFindServices('dataSourceReachabilityService');
+
+    // Get common payload params and pull environment info
+    const requestContext = await this.payload.object('requestContext');
+    // If you specify an id, you can’t specify a status filter
+    const forceCheck = await this.payload.boolean('forceCheckAll');
+    const dsAccountIds = await this.payload.object('dsAccountIds');
```

Comment:
Do you know how many ids we might have?  With StepFunction we can only use 32K (which we share with all the other payload items)?

Reply:
Good point. Although for now we can bypass that scenario by adding a guard to only bulk check for a max of 1000 account Ids. We can add chunking eventually.